sia/src/encoding/v1_encoding.rs

```diff
@@ -0,0 +1,234 @@
+use crate::encoding::{Error, Result};
```

Comment:
Kind of disappointing duplication, but I could not figure out a clean way to support both through a single trait. Trying to use generics to support both was a big PITA.